e2e/test/scenarios/visualizations-tabular/smartscalar-trend.cy.spec.js

```diff
@@ -181,6 +191,26 @@ describe("scenarios > visualizations > trend chart (SmartScalar)", () => {
     // add a suffix
     cy.findByLabelText("Add a suffix").click().type(" ! cool").blur();
     cy.findByTestId("scalar-container").findByText("Woah: 68’800.0000% ! cool");
+
+    // scalar.compact_primary_number setting
+    cy.findByTestId("chartsettings-sidebar").within(() => {
+      cy.findByText("Data").click();
+      cy.findByText("Count").click();
+    });
+    popover().findByRole("option", { name: "Mega Count" }).click();
+    cy.findByTestId("chartsettings-sidebar").findByText("Display").click();
+
+    cy.findByTestId("scalar-container").findByText("3,440,000");
+
+    cy.findByTestId("chartsettings-sidebar")
+      .findByLabelText("Compact number")
+      .click();
+    cy.findByTestId("scalar-container").findByText("3.4M");
+
+    cy.findByTestId("chartsettings-sidebar")
+      .findByLabelText("Compact number")
+      .click();
+    cy.findByTestId("scalar-container").findByText("3,440,000");
```

Comment:
With the compact setting being moved to `computeTrend`, can we add a check to validate that the comparison value is compacted as well?

Reply:
Good call, thanks! Added in [`9238b66`](https://github.com/metabase/metabase/pull/36992/commits/9238b669078b13148db3d07088c8ba4a70670e7b)